battleship.py

```diff
@@ -18,55 +24,108 @@ def clear_grid():
+                        if length >= 5:
+                            fifth = [first[0] + 4, first[1]]
+            else:
+                third = [-1, -1]
+                fourth = [-1, -1]
+                fifth = [-1, -1]
+                first = [random.randint(1, grid_size), random.randint(1, (grid_size - length + 1))]
+                second = [first[0], first[1] + 1]
+                if length >= 3:
+                    third = [first[0], first[1] + 2]
+                    if length >= 4:
+                        fourth = [first[0], first[1] + 3]
+                        if length >= 5:
+                            fifth = [first[0], first[1] + 4]
+            new_ship = [first, second, third, fourth, fifth]
+            for k in new_ship:
+                if check_ships(k, ships):
+                    valid_ship.append(k)
+                else:
+                    check_collision += 1
+            if check_collision == 0:
+                ships.extend(valid_ship)
+                collision = False
     return ships
 
 
 def get_target():
-    choice = input("Where would you like to aim? (example: '12,3')")
-    if not (choice.split(','))[0].isdigit() or not (choice.split(','))[1].isdigit() \
-            or 1 <= int((choice.split(','))[0]) > grid_size or 1 <= int((choice.split(','))[1]) > grid_size:
-        print("Your coordinate was invalid. Please follow the example and try again.")
-        get_target()
+    choice = input("\nWhere would you like to aim? (example: '12,3') \n")
+    try:
+        int(choice)
+    except ValueError:
+        if not (choice.split(','))[0].isdigit() or not (choice.split(','))[1].isdigit() \
+                or 1 <= int((choice.split(','))[0]) > grid_size or 1 <= int((choice.split(','))[1]) > grid_size:
+            target = 0
+            print("Invalid coordinate. Please try again.")
+            return target
+        else:
+            choice_list = choice.split(',')
+            target = []
+            for i in choice_list:
+                target.append(int(i))
+            target = [target[1], target[0]]
+            return target
     else:
-        choice_list = choice.split(',')
-        target = []
-        for i in choice_list:
-            target.append(int(i))
-        target = [target[1], target[0]]
-        return target
+        print("You need TWO numbers. Try again.")
+        return 0
 
 
 def check_availability(target, grid):
-    if grid[(target[0] - 1)][(target[1] - 1)] == "0":
+    if grid[(target[0] - 1)][(target[1] - 1)] == "-":
         return True
     else:
         return False
 
 
 def check_hit(target, ships):
-    if target in ships["carrier"] or target in ships["cruiser"]:
+    if target in plrs[plr]["ships"]:
         return True
     else:
         return False
 
 
-def fire(target, grid):
+def fire(target, grid, ships):
     if check_availability(target, grid):
-        if check_hit(target, grid):
+        if check_hit(target, ships) == True:
```

Comment:
Pro tip:
When you are checking if something is true or false you don't have to say ``` == True``` etc. you can just do 
``` if check_hit(target, ships):```
or 
``` if not check_hit(target, ships):```

It's the same thing but it's just shorter and more elegant.